analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.PatternMatching.cs

```diff
@@ -287,5 +288,246 @@ public void DeclarationPattern_Var_Deconstruction_DoesNotSetConstraint()
             validator.ValidateTag("D", x => x.Should().BeNull());
             validator.ValidateTag("End", x => x.HasConstraint(TestConstraint.First).Should().BeTrue());
         }
+
+        [DataTestMethod]
+        [DataRow("objectNotNull", "null", false)]
+        [DataRow("objectNotNull", "1", null)]
+        [DataRow("objectNotNull", @"""""", null)]
+        [DataRow("objectNotNull", "true", null)]
+        [DataRow("objectNotNull", "false", null)]
+        [DataRow("objectNull", "null", true)]
+        [DataRow("objectNull", "1", null)]
+        [DataRow("objectNull", @"""""", null)]
+        [DataRow("objectNull", "true", null)]
+        [DataRow("objectNull", "false", null)]
+        [DataRow("objectUnknown", "null", null)]
+        [DataRow("objectUnknown", "1", null)]
+        [DataRow("objectUnknown", @"""""", null)]
+        [DataRow("objectUnknown", "true", null)]
+        [DataRow("objectUnknown", "false", null)]
+        [DataRow("nullableBoolTrue", "true", true)]
+        [DataRow("nullableBoolTrue", "false", false)]
+        [DataRow("nullableBoolFalse", "true", false)]
+        [DataRow("nullableBoolFalse", "false", true)]
+        [DataRow("nullableBoolNull", "true", null)]  // FN. Should be false.
+        [DataRow("nullableBoolNull", "false", null)] // FN. Should be false.
+        [DataRow("nullableBoolUnknown", "true", null)]
+        [DataRow("nullableBoolUnknown", "false", null)]
+        public void ConstantPatternSetBoolConstraint(string variableName, string isPattern, bool? expectedBoolConstraint)
+        {
+            const string variableDeclarations = @"
+var objectNotNull = new object();
+var objectNull = (object)null;
+var objectUnknown = Unknown<object>();
+var nullableBoolTrue = (bool?)true;
+var nullableBoolFalse = (bool?)false;
+var nullableBoolNull = (bool?)null;
+var nullableBoolUnknown = Unknown<bool?>();
```

Comment:
Let's merge all of these (from this test and tests below) to the `ValidateSetBoolConstraint`. There's no harm if the method contains more than it uses.